platform/core/Makefile

```diff
@@ -0,0 +1,53 @@
+export_dir_ios = export/ios
+ios_archs := aarch64-apple-ios aarch64-apple-ios-sim x86_64-apple-ios
+ios_targets := iPhone simulator
+build_type := $(if $(filter $(MODE), release),release,debug)
+
+build_ios_xcframework:
+# Ensure correct OS is used
+ifeq ($(OS),Windows_NT)
+	@echo "xcframework can be built only in macOS"
+	exit 1
+else
+ifneq ($(shell uname -s),Darwin)
```

Comment:
Wouldn't it be enough to test that the value of `$(OS)` is the correct one? I understand that you need to check first for windows because `uname` doesn't work there, but $(OS) will probably have a value for OSX which you can check directly.

Reply:
From my tests `OS` is empty when running on macOS. Didn't find other way to achieve this.

Reply:
Well, that's a value :-) Unless it's the same in Linux.

Reply:
I think it is the same :-), based on my research.

Reply:
In that case, there would still be a corner case where an operating system is not Windows_NT but still can't use uname. It probably can be dismissed as impossible or extremely unlikely.